File: ruby/command_injection/file_operations.rb
# VULN 1: system() with string concatenation - file format conversion
def convert_file_format(input_file, output_format)
  system("convert " + input_file + " output." + output_format)
end

# VULN 2: Backtick execution with string interpolation - create zip archive
def create_archive(directory, archive_name)
  `zip -r #{archive_name}.zip #{directory}`
end

# VULN 3: exec() with string interpolation - set file permissions
def set_file_permissions(filepath, permissions)
  exec("chmod #{permissions} #{filepath}")
end

File: java/command_injection/FileOperations.java
import java.io.*;

public class FileOperations {
    // VULN 1: Runtime.exec with string concatenation - file conversion
    public void convertFileFormat(String inputFile, String outputFormat) throws IOException {
        Runtime.getRuntime().exec("convert " + inputFile + " output." + outputFormat);
    }

    // VULN 2: ProcessBuilder sh -c with string concatenation - create zip archive
    public void createArchive(String directory, String archiveName) throws IOException {
        ProcessBuilder pb = new ProcessBuilder("sh", "-c", "zip -r " + archiveName + ".zip " + directory);
        pb.start();
    }

    // VULN 3: Runtime.exec array form with string concatenation - chmod
    public void setFilePermissions(String filepath, String permissions) throws IOException {
        String[] cmd = {"sh", "-c", "chmod " + permissions + " " + filepath};
        Runtime.getRuntime().exec(cmd);
    }
}
